tests/Doctrine/Tests/ORM/QueryTest.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace Doctrine\Tests\ORM;
+
+use Doctrine\Tests\Models\CMS\CmsUser;
+use Doctrine\Tests\OrmTestCase;
+
+/**
+ * Test case for the Query class
+ *
+ * @author Konstantin Kuklin <konstantin.kuklin@gmail.com>
+ */
+class QueryTest extends OrmTestCase
+{
+    /**
+     * @var \Doctrine\ORM\EntityManager
+     */
+    private $_em;
+
+    protected function setUp()
+    {
+        $this->_em = $this->_getTestEntityManager();
+    }
+
+    /**
+     * setResultCacheProfile should works correct with null
+     */
+    public function testSetResultCacheProfile()
+    {
+        $query = $this->_em->createQuery('SELECT u FROM ' . CmsUser::class . ' u');
+        $result = $query->setResultCacheProfile(null);
+        self::assertEquals($result, $query);
```

Comment:
I think we should be using `$this` instead of `self::` like the rest of the code base right?

Reply:
Strictly speaking, the `assert*` methods in PHPUnit are static, so they should be called using `self::assert*`, not `$this->assert*`. IIRC @Ocramius and @lcobucci have started changing these in new tests and for v3.

Reply:
Shouldn't we assert that the cache profile was NOT configured?